src/memgraph.cpp

```diff
@@ -65,9 +65,13 @@ void InitFromCypherlFile(memgraph::query::InterpreterContext &ctx, memgraph::dbm
   std::string line;
   while (std::getline(file, line)) {
     if (!line.empty()) {
-      auto results = interpreter.Prepare(line, {}, {});
-      memgraph::query::DiscardValueResultStream stream;
-      interpreter.Pull(&stream, {}, results.qid);
+      try {
+        auto results = interpreter.Prepare(line, {}, {});
+        memgraph::query::DiscardValueResultStream stream;
+        interpreter.Pull(&stream, {}, results.qid);
+      } catch (const memgraph::query::QueryRuntimeException &e) {
+        spdlog::error("Error while executing init file: {}. The rest of the queries will be run normally.", e.what());
```

Comment:
Before we would throw exception, not handle it and hence Memgraph startup would fail. This is not good, check the issue why. If we do MG_ASSERT then this is the same as not catching exception.

Reply:
Okay, so we trying to fix the already added user throw, but with this implementation we would ignore any failed query, not just the user one. I wouldn't want to have a data related query fail and for us to just ignore it.
I think we should specify the exception, throw something like UserAlreadyExists and catch only that. In all other cases, I would assert.
Another option would be to clear auth data before and assert for any throw. Is there a scenario where we would want to have some users from the preserved auth data and some from the init file?

Reply:
And in any case, it's better to catch the exception and do MG_ASSERT, than to just throw.

Reply:
1. Why is assert better than throwing?
> Another option would be to clear auth data before and assert for any throw. Is there a scenario where we would want to have some users from the preserved auth data and some from the init file?
The error would happen if you close Memgraph and turn it on again
> I think we should specify the exception, throw something like UserAlreadyExists and catch only that. In all other cases, I would assert.
That's an option yes

Reply:
> Why is assert better than throwing?
You are right that in `MG_ASSERT` we call `std::terminated`, which is the same thing called when the program encounters an uncaught exception. 
However, stylistically we have always explicitly used `MG_ASSERT` to indicate an unrecoverable failure.
Having the same error behavior throughout the code is also beneficial, because in the future we could change the `MG_ASSERT` and have all Memgraph specific errors work in the same way.

> The error would happen if you close Memgraph and turn it on again
I think we shouldn't populate auth from disk if the point of using the file to init is to have Memgraph in the exact state as defined by the file.
The current fix wouldn't work if I opened Memgraph, created users, closed Memgraph and tried to initialize Memgraph with a random file. We would get users from the previous session, even though they are not defined in the init file used.

Reply:
So what is your suggestion?

Reply:
If init-file is set, delete all existing auth things

Reply:
@gitbuda any suggestions?
I think we shouldn't do any recovery (auth, replica, etc.) in case we are using a file to initialize.

Reply:
By recovery, you mean recovery of what?

Reply:
Ups, sorry, meant replication recovery.

Reply:
Recovery of durable data. From the various subsystems we have, like authentication, multi-tenancy, replication and so on.

Reply:
Aha so if the `--init-file` is set you wouldn't recover authentication I assume?
And if the `--init-file-data` is set then you wouldn't do a recovery of multi-tenancy, data, replication and so on?

Reply:
Yes, but we have to make sure that all of the data not recovered can be dumped into a file.
Multi-tenancy is not supported at the moment, replication neither probably.
But I think that ignoring all durable data when initializing from a file would be the correct solution.